### PR TITLE
chore: Add matrix for phpunit with macos runner 

### PR DESCRIPTION
**Pull request by TheoD02**
Adding macOS to runner to ensure all test pass with this system

### CODE REVIEW

**Comment by lyrixx** [comment]
> @lyrixx Do the `Build Castor phar for Linux` a

This is weird it does not pass

> Compile Custom Built PHP along Castor phar for Linux

let's skip it

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,11 +47,13 @@ jobs:
...
   phpunit:
-    name: "PHPUnit on ${{ matrix.php }} / Castor from ${{ matrix.castor.method }}"
-    runs-on: ubuntu-latest
+    name: "PHPUnit on ${{ matrix.os }} - ${{ matrix.php }} / Castor from ${{ matrix.castor.method }}"
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
+        os: [ ubuntu-latest, macos-latest ]
+        php: [ "8.1", "8.2", "8.3" ]
```
Comment (lyrixx) [inline .github/workflows/ci.yml:56]: That's too much consumption. The matrix is already quite big. Let's test on linux 8.1 ... 8.3, and only 8.1 on mac

**Comment by TheoD02** [comment]
How do you manage with this case when macOS has a different output from Unix ?

https://github.com/jolicode/castor/actions/runs/8554181097/job/23438948411?pr=377#step:9:3690

**Comment by lyrixx** [comment]
@TheoD02 I don't know 🙃 ; This is why I did work on this yet. And I don't have 
bandwidth ATM to help you. Sorry. Feel free to give up :) This is not a big deal for now

**Comment by TheoD02** [comment]
Okay I'll close the PR for now, I'll try on the side when I have some time, if I have a solution I'll propose 😄

**Pull request closed without merging**